### PR TITLE
Add subzero regression testing initiated from GUI jar

### PR DESCRIPTION
**Pull request by syncom**
Currently use a fixed dev wallet, numbered 1492, and auto generate test vectors for to-be-signed transactions.
Add utils for test vector generation.
Add HOWTO document for subzero regression testing. The document explains how to create and run test vectors for subzero core regression test.
This version only adds 255 valid transactions and 1 invalid transaction.
It does not yet handle HSM backed subzero test, and will deal with this part in a later commit.

### CODE REVIEW

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java [hunk 1]
```diff
@@ -180,4 +203,109 @@ private void uiTest() throws Exception {
...
+    public IllegalTxSignTestFileFormatException(String msg) {
+      super(msg);
+    }
+  }
+
+  private void signTxTest() throws Exception {
+
+    // TODO: implement HSM test
+    if (nCipher) {
+      System.out.println("Transaction signing regression test not yet implemented for nCipher");
```
Comment (oreparaz) [inline java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java:217]: throw not implemented exception?

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java [hunk 2]
```diff
@@ -117,8 +136,12 @@ private void interactive() throws Exception {
 
         CommandRequest commandRequest = CommandRequest.parseFrom(proto);
 
+        System.out.println("print out commandRequest:");
+        System.out.println(commandRequest.toString());
```
Comment (oreparaz) [inline java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java:140]: is this needed? or does it come from your testing?

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java [hunk 3]
```diff
@@ -180,4 +203,109 @@ private void uiTest() throws Exception {
...
+
+        CommandRequest commandRequest = CommandRequest.parseFrom(proto);
+
+        CommandResponse response =
+            CommandHandler.dispatch(this, new InternalCommandConnector(hostname, port),
+                commandRequest);
+
+        String encoded = base64().encode(response.toByteArray());
+
+        if (encoded.equals(expectedResponse)) {
```
Comment (oreparaz) [inline java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java:274]: suggest simplifying logic here if possible, maybe erroring directly on mismatch?

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java [hunk 4]
```diff
@@ -180,4 +203,109 @@ private void uiTest() throws Exception {
...
+    // TODO: implement HSM test
+    if (nCipher) {
+      System.out.println("Transaction signing regression test not yet implemented for nCipher");
+      return;
+    }
+
+    // Passed and failed test cases, for valid and invalid test vectors.
+    // For a valid test vector, test passes (ok) if and only if subzero response matches
+    // expected response.
+    // For an invalid test vector, test passes (ok) if and only if subzero response does not
```
Comment (oreparaz) [inline java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java:224]: had a little bit of trouble understanding "invalid" test vectors

**Review by oreparaz — APPROVED** [review]
Nice work @syncom !

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java (hunk 2 above)
Comment (syncom) [reply java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java:140]: Good catch. This should be cleaned up.

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java (hunk 3 above)
Comment (syncom) [reply java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java:274]: We clarified the cases of "valid" and "invalid" test vectors, which are corresponding to positive and negative test cases. Will keep it as is for now.

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java (hunk 4 above)
Comment (syncom) [reply java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java:224]: I will try clarifying a little bit further in the comment in the upcoming commit.

**Review by oreparaz — APPROVED** [review]

**Pull request merged**